silverback/_cli.py

```diff
@@ -192,35 +198,121 @@ def list_clusters(platform: PlatformClient, workspace: str):
+
+    if cluster_name:
+        click.echo(f"name: {cluster_name}")
+        click.echo(f"slug: {cluster_slug or cluster_name.lower().replace(' ', '-')}")
+
+    elif cluster_slug:
+        click.echo(f"slug: {cluster_slug}")
+
+    cluster = workspace_client.create_cluster(
+        cluster_name=cluster_name,
+        cluster_slug=cluster_slug,
+    )
+    click.echo(f"{click.style('SUCCESS', fg='green')}: Created '{cluster.name}'")
+
+    if cluster.status == ResourceStatus.CREATED:
+        click.echo(
+            f"{click.style('WARNING', fg='yellow')}: To use this cluster, "
+            f"please pay via `silverback cluster pay create {workspace}/{cluster_slug}`"
+        )
+
+
+@cluster.group(cls=SectionedHelpGroup, section="Platform Commands (https://silverback.apeworx.io)")
+def pay():
+    """Pay for CLUSTER with Crypto using ApePay streaming payments"""
+
+
+@pay.command(name="create", cls=ConnectedProviderCommand)
+@account_option()
+@click.argument("cluster_path")
 @click.option(
     "-t",
     "--tier",
-    default=ClusterTier.PERSONAL.name,
+    default=ClusterTier.PERSONAL.name.capitalize(),
     metavar="NAME",
+    type=click.Choice(
+        [
+            ClusterTier.PERSONAL.name.capitalize(),
+            ClusterTier.PROFESSIONAL.name.capitalize(),
+        ]
+    ),
     help="Named set of options to use for cluster as a base (Defaults to Personal)",
 )
 @click.option(
     "-c",
     "--config",
     "config_updates",
+    metavar="KEY VALUE",
     type=(str, str),
     multiple=True,
-    help="Config options to set for cluster (overrides value of -t/--tier)",
+    help="Config options to set for cluster (overrides values from -t/--tier selection)",
+)
+@click.option("--token", metavar="ADDRESS", help="Token Symbol or Address to use to fund stream")
+@click.option(
+    "--amount",
+    "token_amount",
+    metavar="VALUE",
+    callback=token_amount_callback,
+    default=None,
+    help="Token amount to use to fund stream",
+)
+@click.option(
+    "--time",
+    "stream_time",
+    metavar="TIMESTAMP or TIMEDELTA",
+    callback=timedelta_callback,
+    default=None,
+    help="Time to fund stream for",
 )
-@click.argument("workspace")
 @platform_client
-def new_cluster(
+def create_payment_stream(
     platform: PlatformClient,
-    workspace: str,
-    cluster_name: str | None,
-    cluster_slug: str | None,
+    network: NetworkAPI,
+    account: AccountAPI,
+    cluster_path: str,
     tier: str,
     config_updates: list[tuple[str, str]],
+    token: ContractInstance | None,
+    token_amount: int | None,
+    stream_time: timedelta | None,
 ):
-    """Create a new cluster in WORKSPACE"""
+    """
+    Create a new streaming payment for a given CLUSTER
 
-    if not (workspace_client := platform.workspaces.get(workspace)):
-        raise click.BadOptionUsage("workspace", f"Unknown workspace '{workspace}'")
+    NOTE: This action is irreversible!
```

Comment:
question: Is it?

Reply:
Once the stream is created, the stream owner cannot cancel it for 1 hour, giving time for the infra to fully deploy without allowing the stream parameters to update, causing us issues

I made that more apparent w 7dbe052